ovn-tester/ovn_stats.py

```diff
@@ -13,10 +14,16 @@ def timeit(func):
     @functools.wraps(func)
     def _timeit(*args, **kwargs):
         start = time.perf_counter()
-        value = func(*args, **kwargs)
-        duration = time.perf_counter() - start
-        add(func.__name__, duration)
-        return value
+        failed = False
+        value = None
+        try:
+            value = func(*args, **kwargs)
+        except ovn_utils.OvnTestException:
+            failed = True
+        finally:
+            duration = time.perf_counter() - start
+            add(func.__name__, duration, failed)
+            return value
     return _timeit
```

Comment:
This needs to be changed to:
```
def timeit(func):
    @functools.wraps(func)
    def _timeit(*args, **kwargs):
        start = time.perf_counter()
        failed = False
        value = None
        try:
            value = func(*args, **kwargs)
        except ovn_utils.OvnTestException:
            failed = True
        finally:
            duration = time.perf_counter() - start
            add(func.__name__, duration, failed)
        return value
    return _timeit
```
Otherwise a non-OvnTestException will never be re-raised, inadvertently hiding bugs, e.g., TypeError.

Reply:
Reference: https://docs.python.org/3/tutorial/errors.html#defining-clean-up-actions
```
If the finally clause executes a break, continue or return statement, exceptions are not re-raised.
```